src/query.rs

```diff
@@ -0,0 +1,104 @@
+use pest::Parser;
+use std::fmt;
+
+use crate::index::{get_packet_index, Index};
+
+pub fn run_query(root: &str, query: String) -> Result<String, QueryError> {
+    let index = match get_packet_index(root) {
+        Ok(index) => index,
+        Err(e) => {
+            return Err(QueryError::EvalError(format!(
+                "Could not build outpack index from root at {}: {:?}",
+                root, e
+            )))
+        }
+    };
+    let parsed: QueryNode = parse_query(&query)?;
+    eval_query(index, parsed)
+}
+
+#[derive(Parser)]
+#[grammar = "query.pest"]
+pub struct QueryParser;
+
+enum QueryNode<'a> {
+    Latest,
+    Lookup(&'a str),
+}
+
+#[derive(Debug, Clone)]
+pub enum QueryError {
+    ParseError(Box<pest::error::Error<Rule>>),
```

Comment:
maybe worth a comment about why a Box is needed here? I don't really get it :sweat_smile:

Reply:
Thanks, added that